TOOLS/gen-interface-changes.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+
+# Generate a new interface-changes.rst based on the entries in
+# the interface-changes directory.
+
+#
+# This file is part of mpv.
+#
+# mpv is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# mpv is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with mpv.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import pathlib
+import sys
+
+def add_new_entries(docs_dir, out):
+    changes_dir = pathlib.Path(docs_dir) / "interface-changes"
+    files = [f for f in pathlib.Path(changes_dir).glob("*.txt") if f.is_file()]
+    for file in files:
```

Comment:
I would prefer for this list to be sorted, depending on the file system this will produce different results.

Additionally to have ordered interface-changes.rst. I would suggest the file names in the `
DOCS/interface-changes` to be in `13587-bla.txt` or even `13587.txt` or `13587`. There have to be guideline for filename naming to avoid big mess in interface-changes file.

P.S. I don't thinks `example.txt` is that useful, after a while we will see real examples in this folder.

Reply:
Is sorting even really useful here? I mean adding .sort() is not a huge deal, but I'm not sure if expecting the files in this directory to be in some sort of order is a worthwhile endeavor. As far as `example.txt` goes, I don't think it's useful either but there has to be at least some placeholder in this directory. Could move the instructions in here again or something else entirely.

Reply:
Sometimes interface changes may logically depend on previous changes that were not done in the same commit. Random shuffling the changelog seems weird to me.

Reply:
Something like that should just be added to the existing file as an additional line imo. I mean I'm only saying this because I don't see enforcing a number scheme or something similar as particularly practical. Like if a PR is merged are the other ones supposed to +1 to their existing number or something.

Reply:
Ok, let's do it this way. My only concert is that not all logically related changes my be in the same PR, so some interface changes may feel odd when read "out of order". Not a big deal, tho.